lightning/src/onion_message/offers.rs

```diff
@@ -47,7 +47,7 @@ pub trait OffersMessageHandler {
 	/// [`OnionMessenger`]: crate::onion_message::messenger::OnionMessenger
 	fn handle_message(
 		&self, message: OffersMessage, context: Option<OffersContext>, responder: Option<Responder>,
-	) -> Option<(OffersMessage, ResponseInstruction)>;
+	) -> Option<(OffersMessage, MessageSendInstructions)>;
```

Comment:
`Responder` is kinda useless then if you don't need it to construct instructions.

Reply:
How about keeping `ResponseInstructions` for the handlers and converting it into `MessageSendInstructions` inside `handle_onion_message_response? Then there is no wrapping.

Reply:
Not really identical. `ResponseInstructions` would have only two variants and would have a `BlindedMessagePath` instead of a `Destination`.

Reply:
The motivation is to enforce the correct behavior of only using the reply path for the response. And it prevents users from doing something wrong like manually creating a `MessageSendInstructions` to return when the `responder` is `None`.

Reply:
> You mean only using the reply path we were given in the message to respond to that message? Is the concern there that someone would accidentally use that reply path to send another message or to receive some message (treating it as our own reply path)? I'm not sure how likely that is, though maybe leaving the "return an identical type from `respond` as the method returns" thing would help there (without having to define yet another type)?

It's more so having them be confused about how to create the return value and think they need to create a `Destination` themselves (and thus either give it a node id or create a blinded path) when they should use the `Responder` to create it. If the return type is `ResponseInstructions` and the only way to create one is through `Responder`, then they can't screw it up.

Though I guess `ResponseInstructions` can't have a `OnionMessageResponse` (because it has a type parameter). So instead of a nearly identical enum, it would need to be a struct with private constructors:

```rust
pub struct ResponseInstructions {
    destination: BlindedMessagePath,
    include_reply_path: bool,
}
```

> I mean in order to do so they'd have to fill in the `Destination` with something, which seems like it'd be pretty hard to get wrong?

The point is they shouldn't be doing that at all, but our interface allows for it.